bigtable.grpc.async.TestBulkRead: testMissingResponses failed

This test failed!

To configure my behavior, see [the Build Cop Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/master/packages/buildcop).

If I'm commenting on this issue too often, add the `buildcop: quiet` label and
I will stop commenting.

---

commit: 6bff6e59ba523e241e9e03394aa6f6b1c9aec32b
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/b1ecf01c-9194-43e5-a6f0-473e55cbbcf3), [Sponge](http://sponge2/b1ecf01c-9194-43e5-a6f0-473e55cbbcf3)
status: failed
<details><summary>Test output</summary><br><pre>java.lang.AssertionError: expected null, but was:<FlatRow{rowKey=<ByteString@58d8059b size=8 contents="83206310">, cells=[Cell{family=family, qualifier=<ByteString@1eb08757 size=0 contents="">, timestamp=1606811037789, value=<ByteString@1eb08757 size=0 contents="">, labels=[]}]}>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotNull(Assert.java:756)
	at org.junit.Assert.assertNull(Assert.java:738)
	at org.junit.Assert.assertNull(Assert.java:748)
	at com.google.cloud.bigtable.grpc.async.TestBulkRead.testMissingResponses(TestBulkRead.java:168)
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:59)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:56)
	at org.junit.internal.runners.statements.InvokeMethod.evaluate(InvokeMethod.java:17)
	at org.junit.internal.runners.statements.RunBefores.evaluate(RunBefores.java:26)
	at org.mockito.internal.junit.JUnitSessionStore$1.evaluateSafely(JUnitSessionStore.java:54)
	at org.mockito.internal.junit.JUnitSessionStore$1.evaluate(JUnitSessionStore.java:42)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.BlockJUnit4ClassRunner$1.evaluate(BlockJUnit4ClassRunner.java:100)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:366)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:103)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:63)
	at org.junit.runners.ParentRunner$4.run(ParentRunner.java:331)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:79)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:329)
	at org.junit.runners.ParentRunner.access$100(ParentRunner.java:66)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:293)
	at org.junit.runners.ParentRunner$3.evaluate(ParentRunner.java:306)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:413)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:365)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:273)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:238)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:159)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:384)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:345)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:126)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:418)
</pre></details>

```
[ERROR] Tests run: 4, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.334 s <<< FAILURE! - in com.google.cloud.bigtable.grpc.async.TestBulkRead
[ERROR] testMissingResponses(com.google.cloud.bigtable.grpc.async.TestBulkRead)  Time elapsed: 0.293 s  <<< FAILURE!
java.lang.AssertionError: expected null, but was:<FlatRow{rowKey=<ByteString@58d8059b size=8 contents="83206310">, cells=[Cell{family=family, qualifier=<ByteString@1eb08757 size=0 contents="">, timestamp=1606811037789, value=<ByteString@1eb08757 size=0 contents="">, labels=[]}]}>
	at org.junit.Assert.fail(Assert.java:89)
	at org.junit.Assert.failNotNull(Assert.java:756)
	at org.junit.Assert.assertNull(Assert.java:738)
	at org.junit.Assert.assertNull(Assert.java:748)
	at com.google.cloud.bigtable.grpc.async.TestBulkRead.testMissingResponses(TestBulkRead.java:168)
       ...
```

--- a/bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkRead.java
+++ b/bigtable-client-core-parent/bigtable-client-core/src/test/java/com/google/cloud/bigtable/grpc/async/TestBulkRead.java
@@ -35,6 +35,7 @@ import java.util.Collections;
 import java.util.HashSet;
 import java.util.Iterator;
 import java.util.List;
+import java.util.Random;
 import java.util.Set;
 import java.util.concurrent.ExecutorService;
 import java.util.concurrent.Future;
@@ -60,6 +61,8 @@ public class TestBulkRead {
       new BigtableTableName("projects/SomeProject/instances/SomeInstance/tables/SomeTable");
   private static final RequestContext CONTEXT =
       RequestContext.create("SomeProject", "SomeInstance", "");
+
+  private static final Random random = new Random(2128506);
 
   @Rule public final MockitoRule mockitoRule = MockitoJUnit.rule();
 
@@ -158,7 +161,7 @@ public class TestBulkRead {
                 Query request = invocation.getArgument(0, Query.class);
                 List<ByteString> rowKeysList =
                     new ArrayList<>(request.toProto(CONTEXT).getRows().getRowKeysList());
-                missing.add(rowKeysList.remove((int) Math.random() * rowKeysList.size()));
+                missing.add(rowKeysList.remove(random.nextInt(rowKeysList.size())));
                 return createMockScanner(rowKeysList.iterator());
               }
             });
@@ -202,7 +205,7 @@ public class TestBulkRead {
     List<ByteString> rowKeys = new ArrayList<>();
     for (int i = 0; i < count; i++) {
       // Create a random list of keys as input
-      rowKeys.add(ByteString.copyFromUtf8(String.valueOf((int) (Math.random() * 100000000))));
+      rowKeys.add(ByteString.copyFromUtf8(String.valueOf(random.nextInt(100000000))));
     }
     return rowKeys;
   }

